### PR TITLE
Finalise spacefinder changes from last quarter

### PR DESCRIPTION
**Pull request by simonbyford**
## What does this change?

We got the results back from `SpacefinderOkrMegaTest` and found it was a success! This means we can finally abandon the pre-fix logic for each bug, as well as the `SpacefinderOkrMegaTest` definition itself.

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes: https://github.com/guardian/dotcom-rendering/pull/4707

## What is the value of this and can you measure success?

No more broken code. Also, removes the last commercial instance of `bean` forever!

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: tools/__tasks__/commercial/graph/output/standalone.commercial.ts.json [hunk 1]
```diff
@@ -714,23 +704,13 @@
   "../lib/url.ts"
  ],
  "../projects/common/modules/mark-candidates.js": [],
- "../projects/common/modules/on-images-loaded-broken.js": [
-  "../../../../node_modules/bean/bean.js",
```
Comment (simonbyford) [inline tools/__tasks__/commercial/graph/output/standalone.commercial.ts.json:718]: 👋

**Comment by zekehuntergreen** [comment]
We can remove `ab-spacefinder-okr-mega-test` from [ABTestSwitches.scala](https://github.com/guardian/frontend/blob/9c25b4dbaef246939623036e6006029e710eb6ee/common/app/conf/switches/ABTestSwitches.scala#L29) as well.

**Comment by simonbyford** [comment]
> We can remove `ab-spacefinder-okr-mega-test` from [ABTestSwitches.scala](https://github.com/guardian/frontend/blob/9c25b4dbaef246939623036e6006029e710eb6ee/common/app/conf/switches/ABTestSwitches.scala#L29) as well.

Good shout

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.ts [hunk 2]
```diff
@@ -56,17 +53,19 @@ const insertAdAtPara = (
...
+	(maximumAdHeight: number) =>
+	(candidate: SpacefinderItem, lastWinner?: SpacefinderItem): boolean => {
+		if (
+			!lastWinner ||
+			Math.abs(candidate.top - lastWinner.top) - maximumAdHeight >=
+				adSlotClassSelectorSizes.minBelow
+		) {
+			return true;
+		}
+		return false;
```
Comment (chrislomaxjones) [inline static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:67]: Could this just be

```suggestion
return (
	!lastWinner ||
	Math.abs(candidate.top - lastWinner.top) - maximumAdHeight >=
		adSlotClassSelectorSizes.minBelow
);
```

Comment (chrislomaxjones) [inline static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:67]: Could this be simplified slightly?

```suggestion
return (
	!lastWinner ||
	Math.abs(candidate.top - lastWinner.top) - maximumAdHeight >=
		adSlotClassSelectorSizes.minBelow
);
```

Comment (simonbyford) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.ts]: Oooh good shout. 

I ended up with something close to yours but with an explicit early return if `lastWinner` is `undefined`:

https://github.com/guardian/frontend/pull/24938/files#diff-0ca555f0352210b70560a499ac7186955946730493d02e632bb8037e2ef8e9f0R57

This feels a bit easier to follow, what do you think?

**Review by chrislomaxjones — APPROVED** [review]
Very satisfying PR, and great to see the culmination of this work.

**Pull request merged**